livekit-agents/livekit/agents/voice_assistant/voice_assistant.py

```diff
@@ -301,6 +305,16 @@ async def say(
         new_handle.initialize(source=source, synthesis_handle=synthesis_handle)
         self._add_speech_for_playout(new_handle)
 
+    def set_state(self, state: VoiceAssistantState):
+        """Set the current state of the voice assistant"""
+        t = asyncio.create_task(
+            self._room.local_participant.set_attributes(
+                {"voice_assistant.state": state}
+            )
+        )
+        self._state_tasks.add(t)
+        t.add_done_callback(self._state_tasks.remove)
+
```

Comment:
Let's keep this function private. the user should not be using it (renamed to _update_state)
What we can do to avoid state-flickering, is to add a debounce parameter to this function. (delay) 


```suggestion
    def _update_state(self, state: VoiceAssistantState, delay: float = 0.0):
        """Set the current state of the voice assistant"""

        @utils.log_exceptions(logger=logger)
        async def _run_task(delay: float) -> None:
            await asyncio.sleep(delay)

            if self._room.isconnected():
                await self._room.local_participant.set_attributes(
                    {"voice_assistant.state": state}
                )

        if self._update_state_task is not None:
            self._update_state_task.cancel()

        self._update_state_task = asyncio.create_task(_run_task(delay))
```

Reply:
It is set to 0.0 by default, I think we should only need it before switching to the "thinking" state (because VAD can be noisy)